### PR TITLE
New transaction status for proxy

### PR DESCRIPTION
**Pull request by popenta**
Now for the proxy provider in order to get the transaction status we use the `/process-status` route.
The `TransactionOnNetwork.isCompleted` field is now optional and that could impact some people who were dependent on that field. If `isCompleted` is needed call` getTransaction("txHash", withProcessStatus=True)`.

The `isCompleted` field is set only when `getTransaction()` was called using `withProcessStatus = True`.

When calling `getTransaction()` for the proxy provider an optional parameter called `withProcessStatus: bool` can also be provided. If `withProcessStatus` is set to `True` the `/process-status` route is also called. The transaction's status is then replaced with the `process_status`.

### CODE REVIEW

File: src/proxyNetworkProvider.ts [hunk 1]
```diff
@@ -99,15 +99,29 @@ export class ProxyNetworkProvider implements INetworkProvider {
...
+
+        if (withProcessStatus === true) {
+            processStatusPromise = this.getTransactionStatus(txHash);
+        }
+
         let url = this.buildUrlWithQueryParameters(`transaction/${txHash}`, { withResults: "true" });
         let response = await this.doGetGeneric(url);
-        let transaction = TransactionOnNetwork.fromProxyHttpResponse(txHash, response.transaction);
+
+        let transaction: TransactionOnNetwork;
```
Comment (CiprianDraghici) [inline src/proxyNetworkProvider.ts:112]: You can directly return the transaction. This code 
```
let transaction: TransactionOnNetwork;
if (processStatusPromise !== undefined) {
    const processStatus = await processStatusPromise;
    transaction = TransactionOnNetwork.fromProxyHttpResponse(txHash, response.transaction, processStatus);
}
else {
    transaction = TransactionOnNetwork.fromProxyHttpResponse(txHash, response.transaction);
}
```

could be refactored like this:

```
if (processStatusPromise) {
const processStatus = await processStatusPromise;
   return TransactionOnNetwork.fromProxyHttpResponse(txHash, response.transaction, processStatus);
}
return TransactionOnNetwork.fromProxyHttpResponse(txHash, response.transaction);
```

File: src/transactions.ts [hunk 2]
```diff
@@ -35,19 +34,28 @@ export class TransactionOnNetwork {
...
     }
 
-    static fromProxyHttpResponse(txHash: string, response: any): TransactionOnNetwork {
+    static fromProxyHttpResponse(txHash: string, response: any, processStatus?: TransactionStatus | undefined): TransactionOnNetwork {
         let result = TransactionOnNetwork.fromHttpResponse(txHash, response);
         result.contractResults = ContractResults.fromProxyHttpResponse(response.smartContractResults || []);
-        result.isCompleted = new TransactionCompletionStrategyOnProxy().isCompleted(result);
-        // TODO: uniformize transaction status.
+
+        if (processStatus !== undefined) {
```
Comment (CiprianDraghici) [inline src/transactions.ts:41]: `if (processStatus)`

File: src/transactions.ts [hunk 3]
```diff
@@ -35,19 +34,28 @@ export class TransactionOnNetwork {
...
+            result.status = processStatus;
+
+            if (result.status.isSuccessful() || result.status.isFailed()) {
```
Comment (CiprianDraghici) [inline src/transactions.ts:44]: This code 
```
if (result.status.isSuccessful() || result.status.isFailed()) {
  result.isCompleted = true;
}
else {
  result.isCompleted = false;
}
```
could be reduced to:

`result.isCompleted = result.status.isSuccessful() || result.status.isFailed()`

File: src/proxyNetworkProvider.ts (hunk 1 above)
Comment (popenta) [reply src/proxyNetworkProvider.ts:112]: Indeed! Done! Thanks for the suggestion.

File: src/transactions.ts (hunk 3 above)
Comment (popenta) [reply src/transactions.ts:44]: Done! Thanks!

**Review by CiprianDraghici — APPROVED** [review]

**Comment by andreibancioiu** [comment]
Related to: https://github.com/multiversx/mx-chain-proxy-go/pull/386.

**Pull request merged**